specifyweb/specify/tree_extras.py

```diff
@@ -277,7 +277,8 @@ def merge(node, into, agent):
         else:
             child.parent = target
             child.save()
-
+    new_target = model.objects.get(id=into.id)
+    set_fullnames(new_target.definition, node_number_range=[new_target.nodenumber, new_target.highestchildnodenumber])
```

Comment:
What is the purpose of this `new_target` variable? 
Isn't the `into` parameter the exact same value? Or does the `into` parameter at this point reference an old state of the object and this is to ensure the object being passed in as an argument to set_fullnames(...) is 'up to date'?

Reply:
that is because new children can be added (at target node) so need to refetch to get latest highest child node number. Also  now that I'm thinking about it,  we don't need this statement (280 and 281) at all because fullname will be reset as children are added , and that will be done rank by rank so there is no point of this. I'll need to think more about this.

Reply:
fixed. recursion will handle this.